### DISCUSSION TITLE
Prometheus as exporter for SLO-Gen.

### DISCUSSION THREAD

**Discussion by ppothepa24** [google/slo-generator#490] [Q&A]
I am using Prometheus as the exporter for the SLO-Gen. There are very few metrics that are currently being exported. I wanted to customize the exporter.py files to show other relevant metrics(Error Budget Metrics).. Facing some difficulty and any help would be appreciated.

**Comment by lvaylet** [comment]
Hi @ppothepa24, feel free to open a new Pull Request (PR) with your changes and we can discuss them there.

**Reply by lvaylet** [reply]
Hi @ppothepa24, I think I understand what you have in mind. I am just not sure adding the new metrics to `base.py` is relevant, for two reasons.

The first reason is, we might not want these metrics to be exported by default by all the exporters. This could introduce a breaking change for existing users (even if I agree they are just additions, not updates or deletions) and also incur additional ingestion and storage costs (depending on the exporter).

The second reason is, we made the SLO Generator as flexible and configurable as possible through YAML files and CLI options, so everyone can customize their workflow without impacting the other users.

Have you checked [docs/shared/metrics.md](https://github.com/google/slo-generator/blob/84d2d4d0580557f723fa5a18bf14d7dc35932ce4/docs/shared/metrics.md)? From what I understand, you can add a `metrics` block to the configuration of your Prometheus exporter and define the metrics you need there. Then the list of metrics is picked up by the exporter at this step: https://github.com/google/slo-generator/blob/84d2d4d0580557f723fa5a18bf14d7dc35932ce4/slo_generator/exporters/base.py#L81

Does that help?
